src/basic/time-util.c

```diff
@@ -1493,3 +1493,34 @@ int time_change_fd(void) {
 
         return TAKE_FD(fd);
 }
+
+/*
+ * Clamp times *a and *b so that they are in range [min, max]
+ * while still keeping the relative proportion between them.
+ * Returns true if some clamping has been performed, in which case
+ * *a and *b will be updated with the new values.
+ */
+bool clamp_time_interval(usec_t *a, usec_t *b, usec_t min, usec_t max) {
+        assert(min < max);
+        if (*a > *b)
+                return clamp_time_interval(b, a, min, max);
+        if (*a < min) {
+                usec_t new_high = min * (*b) / (*a);
```

Comment:
needs division by zero check

Reply:
btw, this check can be an assert() if we filtered out zero already during parsing and hence know the value can't be zero here anyway...

Reply:
Sounds good.